ludwig/features/category_feature.py

```diff
@@ -187,11 +188,22 @@ def predictions(
         )
         predictions = tf.cast(predictions, dtype=tf.int64)
 
-        return {
-            PREDICTIONS: predictions,
-            PROBABILITIES: probabilities,
-            LOGITS: logits
-        }
+        if self.loss[TYPE] == 'softmax_cross_entropy':
```

Comment:
I woudl invert thsi is, sampled softmax is the special case, if we add more losses in the future they are likely going to be have like regular softmax cross entropy

Reply:
inverted the condition test